gocse/helpers/aws.go

```diff
@@ -0,0 +1,68 @@
+package helpers
+
+import (
+	"log"
+	"os"
+
+	"github.com/fatih/structs"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+// KMS Providers, the credentials used to access the KMS provider
+// See https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#kmsproviders
+
+type awsKMSCredentials struct {
+	AccessKeyID     string `structs:"accessKeyId"`
+	SecretAccessKey string `structs:"secretAccessKey"`
+}
+
+// KMS data key options.
+// See https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#datakeyopts
+
+type awsKMSDataKeyOpts struct {
+	Region   string `bson:"region"`
+	KeyARN   string `bson:"key"`                // the aws key arn
+	Endpoint string `bson:"endpoint,omitempty"` // optional, defaults to kms.<region>.amazonaws.com
+}
+
+type AWS struct {
+	credentials awsKMSCredentials
+	dataKeyOpts awsKMSDataKeyOpts
+}
+
+func (aws *AWS) Credentials() map[string]interface{} {
+	return structs.Map(aws.credentials)
+}
+
+func (aws *AWS) DataKeyOpts() []byte {
+	data, err := bson.Marshal(aws.dataKeyOpts)
+	if err != nil {
+		log.Fatalf("couldn't marshal aws data key opts %v", err)
```

Comment:
Should this be a `panic` instead since you won't be able to continue proper execution without them? Also, would it be useful to suggest a fix such as "Make sure your environment variables are defined in your execution environment. Same comment for all instances.

Reply:
Turns out the driver can consume bson tagged structs so this isn't in the code anymore. I'm not sure this should be a panic since it wouldn't do much good to the end user to print the strace out, and the error itself is more with the marshaling of bson itself.